### PR TITLE
Add new substitute coffee and green tea variants and recipes

### PR DESCRIPTION
**Pull request by XygenSS**
#### Summary

SUMMARY: Content "Add new substitute coffee and green tea variants and recipes"

#### Purpose of change

Currently, the only way to make sweetened tea is to make a fresh tea with sugar, with no way to add sugar to an already brewed cup of tea. This problem persists in several other recipes involving tea and coffee.

Additionally, substitute coffee and green tea has its usefulness expanded with new drink items.

#### Describe the solution

Adds the following items:
- Substitute coffee milk
- Sweetened substitute coffee milk
- Green tea latte

Removes the following recipes:
- Sweetened coffee from raw coffee + water + sugar + boiling action
- Sweetened coffee milk from coffee + milk + sugar
- Sweetened substitute coffee from raw fake coffee + water + sugar + boiling action
- Sweetened tea from raw tea + water + sugar + boiling action
- Sweetened milk tea from tea + milk + sugar

Adds the following recipes:
- Sweetened coffee from coffee + sugar
- Sweetened coffee milk from sweetened coffee + milk
- Sweetened coffee milk from coffee milk + sugar
- Sweetened substitute coffee from substitute coffee + sugar
- Substitute coffee milk from substitute coffee + milk
- Sweetened substitute coffee from sweetened substitute coffee + milk
- Sweetened substitute coffee from substitute coffee milk + sugar
- Sweetened tea from tea + sugar
- Sweetened milk tea from sweetened tea + milk
- Sweetened milk tea from milk tea + sugar
- Green tea latte from green tea + milk + sugar

Changes: 
- Some recipes now take 1 min instead of 1 min 30 sec. Stirring sugar into coffee does not take a minute and a half.
- Some item descriptions changed.

\* "Sugar" is general term for any sweetener item used for crafting

#### Describe alternatives you've considered

- Adding recipe for every possible combinations milk, sugar, tea or coffee and water have been considered so that all drink types can be crafted with only one recipe: Dropped to reduce clutter

- Adding "Green tea with milk" and "Sweetened green tea" : Dropped since it is not necessary

- Splitting relevant coffee and tea recipes into `/json/recipes/food/caffeine` : Dropped since the new file was too short to justify any merit

- Not adding any new items: These were added for consistency and flavor's sake. Open to changes.

#### Testing

Checked all new and changed items via debug learn all recipes and debug hammerspace.

#### Additional context

Under these new recipes, relevant drinks are made incrementally.
To make a cup of sweetened coffee milk, the player must
1. Brew coffee (10mins)
2. Add sugar or milk (1min)
3. Add milk or sugar (1min)

It is no longer possible to directly make sweetened coffee or coffee milk with one recipe. The base coffee must be made first.
Unfortunately these changes will increase total crafting time and overall keypresses.

### CODE REVIEW

**Comment by Lamandus** [comment]
am I getting it right, that you now can't craft sweetened tea in one go? You need first craft the tea and then add sugar?

Why adding the tedium? Why not just adding the extra reciepes?

**Comment by XygenSS** [comment]
@Lamandus 
Having a recipe for every single way of crafting any drink would need about thirty recipes - For example, just a sweetened coffee milk would need four different recipes. Crafting screen clutter is getting worse with every new addition and I was taking the necessary steps to minimize it while allowing maximum flexibility with how the relevant items can be made.

It might possble to condense them into just one or few recipes using AND/OR requirements, but too much overlap is discouraged since it may cause an "... error during recipe finalization that [the] recipe is too complex. In this case, the game may not be able to corectly predict whether it can be crafted."(json_info.md)

An alternative would be to remove the intermediate items (coffee with only milk, coffee with only sugar), but I didn't think it was a good idea.

An argument that following this style every food recipe can then be deconstructed into miniscule steps is a slippery slope and I do not mean to cause additional annoyance in the crafting process.

If you have any better solutions I'd like to hear them.

**Comment by Lamandus** [comment]
@XygenSS the best solution is to leave it as is, to be frank. The chances that you already have brewed tea/Coffee and you have the need to add sugar afterwards is relatively low. In most (I would say 99% the cases), you already have the sugar and the tea leaves ad hand. So why would you need that? In my eyes, the extra steps are not necessary and will leave most players to "not make it" since cooking is already problematic (low morale boost, etc.). And that was not what I intended when I added those reciepes.

File: data/json/items/comestibles/drink.json [hunk 1]
```diff
@@ -744,6 +744,16 @@
...
     "fun": 10
   },
+  {
+    "type": "COMESTIBLE",
+    "id": "milk_coffee_substitute",
+    "copy-from": "coffee_substitute",
+    "name": { "str_sp": "substitute coffee milk" },
+    "spoils_in": "14 hours",
+    "delete": { "flags": [ "NUTRIENT_OVERRIDE" ] },
+    "description": "Imitation coffee brewed from Kentucky coffeetree, this drink has no caffeine, no sweetener, and is still unplesantly bitter even with the added milk.",
```
Comment (LaVeyanFiend) [inline data/json/items/comestibles/drink.json:754]:
```suggestion
    "description": "Imitation coffee brewed from Kentucky coffeetree, this drink has no caffeine, no sweetener, and is still unpleasantly bitter even with the added milk.",
```

File: data/json/items/comestibles/drink.json [hunk 2]
```diff
@@ -1102,9 +1112,22 @@
...
+  {
+    "type": "COMESTIBLE",
+    "id": "milk_tea_green_sweetened",
+    "name": "green tea latte",
+    "copy-from": "tea",
+    "color": "green",
+    "fatigue_mod": 7,
+    "stim": 2,
+    "calories": 80,
+    "description": "A popular sweet drink, made with green tea, milk, sweetner, and various toppings.  This one is sadly missing the signature tapioca balls, though.",
```
Comment (LaVeyanFiend) [inline data/json/items/comestibles/drink.json:1126]:
```suggestion
    "description": "A popular sweet drink, made with green tea, milk, sweetener, and various toppings.  This one is sadly missing the signature tapioca balls, though.",
```

**Review by LaVeyanFiend — CHANGES_REQUESTED** [review]

**Comment by kevingranade** [comment]
Multi-step recipes are definitely problematic, that's a lot of hassle and knowledge players need just to make tea or coffee.

Let's take a look at this starting with the outcome and working backwards.

We have four base drinks, coffee, coffee substitute, tea, and green tea, though this is subject to expansion.
Each of these allows sweetening and/or milk addition.

So we have:
coffee
coffee with milk
coffee with sugar
coffee with milk and sugar
Repeat for the four drink types and you have a total of 16 recipes.

We need a "beverage sweetener" requirement group, and a "milk or substitute" item group to unify the two variants.

I'm not clear how this is a problematic outcome, as you need even more recipes for the stepwise crafting.

**Comment by XygenSS** [comment]
Yeah, pretty dumb "solution" in retrospect. Thanks for all the comments.

**Pull request closed without merging**